tests/containers/runc_integration.pm

```diff
@@ -0,0 +1,91 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: runc
+# Summary: Upstream runc integration tests
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal qw(select_serial_terminal);
+use utils qw(script_retry);
+use containers::common;
+use containers::bats qw(install_bats switch_to_user delegate_controllers);
+use version_utils qw(is_tumbleweed);
+
+my $test_dir = "/var/tmp";
+my $runc_version = "";
+
+sub run_tests {
+    my %params = @_;
+    my ($rootless, $skip_tests) = ($params{rootless}, $params{skip_tests});
+
+    my $log_file = "runc-" . ($rootless ? "user" : "root") . ".tap";
+
+    assert_script_run "cp -r tests/integration.orig tests/integration";
+    my @skip_tests = split(/\s+/, get_var('RUNC_BATS_SKIP', '') . " " . $skip_tests);
+    script_run "rm tests/integration/$_.bats" foreach (@skip_tests);
+
+    assert_script_run "echo $log_file .. > $log_file";
+    script_run "RUNC=/usr/bin/runc RUNC_USE_SYSTEMD=1 bats --tap tests/integration | tee -a $log_file", 1200;
+    parse_extra_log(TAP => $log_file);
+    assert_script_run "rm -rf tests/integration";
+}
+
+sub run {
+    my ($self) = @_;
+    select_serial_terminal;
+
+    install_bats;
+
+    # Install tests dependencies
+    my @pkgs = qw(git-core glibc-devel-static go iptables jq libseccomp-devel make runc);
+    push @pkgs, "criu" if is_tumbleweed;
+    install_packages(@pkgs);
+
+    delegate_controllers;
+
+    switch_to_user;
+
+    my $test_dir = "/var/tmp";
+    assert_script_run "cd $test_dir";
+
+    # Download runc sources
+    $runc_version = script_output "runc --version  | awk '{ print \$3 }'";
+    script_retry("curl -sL https://github.com/opencontainers/runc/archive/refs/tags/v$runc_version.tar.gz | tar -zxf -", retry => 5, delay => 60, timeout => 300);
+    assert_script_run "cd $test_dir/runc-$runc_version/";
+    assert_script_run "cp -r tests/integration tests/integration.orig";
+
+    # Compile some stuff
+    assert_script_run "make recvtty sd-helper seccompagent fs-idmap memfd-bind pidfd-kill remap-rootfs";
+
+    switch_cgroup_version($self, 2);
+
+    run_tests(rootless => 1, skip_tests => get_var('RUNC_BATS_SKIP_USER', ''));
+
+    select_serial_terminal;
```

Comment:
this is duplicate.

Reply:
> this is duplicate.

Not duplicate because in ` switch_to_user` we switch to an user console.